tests/sentry/api/endpoints/test_organization_teams.py

```diff
@@ -229,11 +233,122 @@ def test_duplicate(self):
+        response = self.get_error_response(
+            self.organization.slug,
+            name="hello world",
+            slug="foobar",
+            set_team_admin=True,
+            status_code=404,
+        )
+        assert response.data == {
+            "detail": "You do not have permission to join a new team as a team admin"
+        }
+
+    @with_feature(["organizations:team-roles", "organizations:team-project-creation-all"])
+    @patch.object(OrganizationTeamsEndpoint, "should_add_creator_to_team", return_value=False)
+    def test_team_admin_not_authenticated(self, mock_creator_check):
+        response = self.get_error_response(
+            self.organization.slug,
+            name="hello world",
+            slug="foobar",
+            set_team_admin=True,
+            status_code=400,
+        )
+        assert response.data == {
+            "detail": "You do not have permission to join a new team as a Team Admin"
+        }
+        mock_creator_check.assert_called_once()
+
+    @with_feature(["organizations:team-roles", "organizations:team-project-creation-all"])
+    def test_team_admin_member_does_not_exist(self):
+        prior_team_count = Team.objects.count()
+
+        # Multiple calls are made to OrganizationMember.objects.get, so in order to only raise
+        # OrganizationMember.DoesNotExist for the correct call, we set a reference to the actual
+        # function then call the reference unless the organization matches the test case
+        get_reference = OrganizationMember.objects.get
+
+        def get_callthrough(*args, **kwargs):
+            if self.organization in kwargs.values():
+                raise OrganizationMember.DoesNotExist
+            return get_reference(*args, **kwargs)
+
+        with patch.object(OrganizationMember.objects, "get", side_effect=get_callthrough):
+            response = self.get_error_response(
+                self.organization.slug,
+                name="hello world",
+                slug="foobar",
+                set_team_admin=True,
+                status_code=403,
+            )
+            assert response.data == {
+                "detail": "You must be a member of the organization to join a new team as a Team Admin",
+            }
+        assert Team.objects.count() == prior_team_count
+
+    @with_feature(["organizations:team-roles", "organizations:team-project-creation-all"])
+    @patch.object(OrganizationMemberTeam.objects, "create", side_effect=IntegrityError)
+    def test_team_admin_org_member_team_create_fails(self, mock_create):
+        prior_team_count = Team.objects.count()
+
+        self.get_error_response(
+            self.organization.slug,
+            name="hello world",
+            slug="foobar",
+            set_team_admin=True,
+            status_code=409,
+        )
+        mock_create.assert_called_once()
+        # check that the created team was rolled back
+        assert Team.objects.count() == prior_team_count
```

Comment:
This test checks that the team is rolled back when `OrganizationMemberTeam` create fails

Reply:
You should also do this for the snowflake error.